### PR TITLE
Add himself as collaborator issue fixed#4735

### LINKED ISSUE #4735: 🐞 Bug One Can Add Himself Again As Collaborator
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

An user can add his own mail as his/her collaborator .

### Steps To Reproduce

1. Log in and go to your dashboard
2. Go to your projects and click any project
3. Click add collaborator and enter own mail and submit.(And see it will be added.)

### Expected Behavior

If an user want to add himself as collaborator proper error message should be shown, and adding should be prevented.

### Screenshots

https://github.com/CircuitVerse/CircuitVerse/assets/104610119/4174cc33-2783-4c6c-83c1-34ad6e243067

### Device Information [optional]

```markdown
- OS:
- Browser:
- version:
```

### Additional context

_No response_

### Are you working on this issue?

Yes

### PR DESCRIPTION
**Pull request by priyam-03**
Fixes #4735

#### Describe the changes you have made in this PR -
When one add collaborator he/she can add his/her own mail previously, now when adding the new array of collaborator I am excluding his/her own email and adding the filtered mail list.
### Screenshots of the changes (If any) -

https://github.com/CircuitVerse/CircuitVerse/assets/104610119/35861afc-d9b1-44ec-9fc0-6f5412b983f7

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema[7.0].define(version: 2023_08_25_140331) do
+ActiveRecord::Schema[7.0].define(version: 2024_02_17_054158) do
```
Comment (vedant-jain03) [inline db/schema.rb:13]: I think this is mistakenly added?

Comment (priyam-03) [reply db/schema.rb:13]: Actually I think after adding mentor_token and other fields the version is changed in my local system.

File: app/controllers/collaborations_controller.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@ def create
     already_present = User.where(id: @project.collaborations.pluck(:user_id)).pluck(:email)
     collaboration_emails = collaboration_params[:emails].grep(Devise.email_regexp)
 
-    newly_added = collaboration_emails - already_present
+    newly_added = collaboration_emails - already_present - [current_user.email]
```
Comment (tachyons) [inline app/controllers/collaborations_controller.rb:23]: Please add rspec tests for this.

Comment (priyam-03) [reply app/controllers/collaborations_controller.rb:23]: @tachyons okay I am adding that.

**Comment by priyam-03** [comment]
There was some mistakes in the pull request so I am closing it

**Pull request closed without merging**